### ISSUE TITLE
Slug control from Elementor

### ISSUE THREAD

**Issue by svipic** [pojome/elementor#7044]
Its gonna solve a huge downside in doing back and forth from Elementor to WP admin panel for changing a slug if needed.
When creating a page, Sometimes there is a need to change a slug of a page before we click for "Publish" button. But right now there is no option for this so we need to publish a page, and then go back to admin and update a slug. It will be very useful for moving this option to Elementor page setting to avoid unnecessary steps.

**Comment by kochhase** [comment]
Hi svipic yes you are right! thats what i have forgott at my closed futurerequest #7041

**Comment by shilo-ey** [comment]
Hi @svipic 
Thank you for your feedback.

We will look into it and update soon.

Thanks.

**Comment by tristanleboss** [comment]
Being able to edit the taonomies from the psot "Settings" panel (where youc an already edit the title, extract, features image, ...) would be a great addition too.

**Comment by jazir5** [comment]
Yes please change this soon. Being able to remove "?elementor_library=" is a necessity to simplify url's for users, makes them easier to remember and natural.

**Comment by Glofu** [comment]
Hello there,
has any solution ever been found?
We are still dealing with weird url there

**Comment by jazir5** [comment]
> Hello there,
> has any solution ever been found?
> We are still dealing with weird url there

https://wordpress.org/plugins/post-type-switcher/

Change the post type to page then change the slug

**Comment by Glofu** [comment]
Hi,
I can't change the type. I need it to be like that because it's an
Elementor template that displays for search results.

Il giorno gio 8 ago 2019 alle ore 19:58 jazir5 <notifications@github.com>
ha scritto:

**Comment by jazir5** [comment]
Can you clarify that? It should still display in search results.

**Comment by Glofu** [comment]
Sorry, I may not have been clear enough actually.
I'm talking about the "search result" page itself. The page you land into
after you search anything in the search bar.
That's why it needs to be a template

Il giorno ven 9 ago 2019 alle ore 20:46 jazir5 <notifications@github.com>
ha scritto:

> Can you clarify that? It should still display in search results.
>

**Comment by bpadonou** [comment]
> > Hello there,
> > has any solution ever been found?
> > We are still dealing with weird url there
> 
> https://wordpress.org/plugins/post-type-switcher/
> 
> Change the post type to page then change the slug. Been using this workaround for months.

Fantastic. It's worked for me. Thanks you

**Comment by stubbi** [comment]
@Glofu running into the same issue, which had also been reported [here](https://wordpress.org/support/topic/custom-archive-page-url-problem/). Could you resolve it? I am using Polylang free, can this be related?

**Comment by 79ho3ein** [comment]
+1

**Comment by Ssendam0101** [comment]
This is still an issue without the workaround of changing the template type to a page. 

The problem from what I can tell is that elementor prefixes archives with ?elementor_library so if you're creating custom taxonomies and expecting to just go to archive templates directly, it will display that as part of the URL. You aren't able to change permalinks in the Elementor Theme builder (at least for archives) but once you change the post type with that plug in to "page" you can, since permalinks are exposed.

From what I can tell, you can't have a template itself use a permalink so the fix here would be for elementor to remove the ?elementor_library prefix which I assume has other issues.

**Comment by atheraries** [comment]
no solution yet?

**Comment by atheraries** [comment]
no solution yet?

**Comment by Theo-jt** [comment]
I cant believe this has not been addressed. Makes Libraries totally useless; looks terribly amateurish not to have any control over the slug. "elementor_library=" in url is terrible!
